nolearn/lasagne/base.py

```diff
@@ -117,6 +146,17 @@ def __init__(
             objective_loss_function = (
                 mse if regression else categorical_crossentropy)
 
+        if hasattr(self, 'train_test_split'):  # BBB
+            warn("The 'train_test_split' method has been deprecated, please "
+                 "use the 'train_split' parameter instead.")
+            train_split = LegacyTrainTestSplit(
+                eval_size=kwargs.pop('eval_size', 0.2))
+
+        if 'eval_size' in kwargs:  # BBB
```

Comment:
You passed `eval_size`, but have no custom subclass with `train_test_split`.  See this test: https://github.com/dnouri/nolearn/blob/feature-train-test-splitter/nolearn/lasagne/tests/test_base.py#L249-L251